crankshaft-engine/src/service/runner/backend/docker.rs

```diff
@@ -59,67 +308,123 @@ impl crate::Backend for Backend {
         "docker"
     }
 
-    fn run(&self, task: Task) -> Result<BoxFuture<'static, Result<NonEmpty<Output>>>> {
+    fn run(
+        &self,
+        task: Task,
+        started: Arc<dyn Fn(usize) + Send + Sync + 'static>,
+        token: CancellationToken,
+    ) -> Result<BoxFuture<'static, Result<NonEmpty<Output>>>> {
         let client = self.client.clone();
         let cleanup = self.config.cleanup();
+        let resources = self.resources;
         let mounts = get_shared_mounts(task.shared_volumes())?;
 
         Ok(async move {
+            // Check to see if we should use the service API for running the task
             let mut outputs = Vec::new();
 
-            for execution in task.executions() {
-                // (1) Create the container.
-                let mut builder = client
-                    .container_builder()
-                    .image(execution.image())
-                    .program(execution.program())
-                    .args(execution.args())
-                    .attach_stdout()
-                    .attach_stderr()
-                    .host_config(HostConfig {
-                        mounts: Some(mounts.clone()),
-                        ..task.resources().map(HostConfig::from).unwrap_or_default()
-                    });
-
-                if let Some(work_dir) = execution.work_dir() {
-                    builder = builder.work_dir(work_dir.to_owned());
-                }
+            if resources.use_service() {
+                let name = task
```

Comment:
This occurs in both parts of the `if`/`else` statement and should just be pulled out in front of it.